File: src/test/java/com/dsobko/credible/test/assertions/HomePageAssertion.java
package com.dsobko.credible.test.assertions;

import com.dsobko.credible.test.pages.HomePage;
import org.fest.assertions.AssertExtension;
import org.slf4j.Logger;
import static org.fest.assertions.api.Assertions.assertThat;
import static org.slf4j.LoggerFactory.getLogger;

public final class HomePageAssertion implements AssertExtension {

    private final Logger log;
    private final HomePage homepage;

    public HomePageAssertion(HomePage homepage) {
        super();
        this.homepage = homepage;
        this.log = getLogger(this.getClass());
    }

    public final HomePageAssertion assertHomePage() {
        this.log.info("Asserting home page elements");
        assertThat(homepage.getPersonalLoansLink().isDisplayed()).isTrue();
        return this;
    }

}

File: src/test/java/com/dsobko/credible/test/assertions/HomePageAssertion.kt
package com.dsobko.credible.test.assertions

import com.dsobko.credible.test.pages.HomePage
import org.fest.assertions.AssertExtension
import org.fest.assertions.api.Assertions.assertThat
import org.slf4j.LoggerFactory.getLogger


class HomePageAssertion(private val homepage: HomePage) : AssertExtension {

    private val log = getLogger(this.javaClass)

    fun assertHomePage(): HomePageAssertion {
        log.info("Asserting home page elements")
        homepage.apply {
            assertThat(personalLoansLink.isDisplayed).isTrue
        }
        return this
    }



}